File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/controller/IngredienteController.java
package controller;

import exception.ingredienteException.IngredienteException;
import models.Ingrediente;
import repository.ingrediente.IngredienteRepository;

import java.io.IOException;
import java.util.List;

import static validator.Validator.validateIngrediente;
import static validator.Validator.validateNumber;

public class IngredienteController{

    private final IngredienteRepository repository;

    public IngredienteController(IngredienteRepository repository) {
        this.repository = repository;
    }

    public List<Ingrediente> getAll() throws Exception {
        return repository.getAll();
    }

    public Ingrediente getById(String id) throws Exception {
        validateNumber(id, new IngredienteException.IngredienteBadRequestException(id), 1);
        Ingrediente ingrediente = repository.getById(Integer.parseInt(id));
        if(ingrediente !=null){
            return ingrediente;
        }else {
            throw new IngredienteException.IngredienteNotFoundException(id);
        }
    }

    public Ingrediente add(Ingrediente entity) throws Exception {
        validateIngrediente(entity);
        Ingrediente ingrediente = repository.add(entity);
        if(ingrediente !=null){
            return ingrediente;
        }else {
            throw new IngredienteException.IngredienteAlreadyExistsException(""+entity.getId());
        }
    }

    public Ingrediente update(Ingrediente entity) {
        return null;
    }

    public Ingrediente remove(Integer id) {
        return null;
    }

    public List<Ingrediente> getByName(String name) {
        return null;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/controller/IngredienteController.kt
package controller

import exception.ingredienteException.IngredienteAlreadyExistsException
import exception.ingredienteException.IngredienteBadRequestException
import exception.ingredienteException.IngredienteNotFoundException
import model.Ingrediente
import mu.KotlinLogging
import repository.Ingrediente.IngredienteRepository
import validator.validate
import validator.validateNumber

class IngredienteController(
    private val ingredienteRepository: IngredienteRepository
) {

    private val logger = KotlinLogging.logger {  }

    fun findByName(name: String): List<Ingrediente> {
        logger.debug { "Controller: Se buscan ingredientes según su nombre." }
        TODO("Not yet implemented")
    }

    fun findById(id: String): Ingrediente {
        logger.debug { "Controller: Se busca un ingrediente según su id." }
        id.validateNumber(IngredienteBadRequestException("id = $id"), 1)
        return ingredienteRepository.findById(id.toInt())?: throw IngredienteNotFoundException(id)
    }

    fun getAll(): List<Ingrediente> {
        logger.debug { "Controller: Se consiguen ingredientes." }
        return ingredienteRepository.getAll()
    }

    fun add(entity: Ingrediente): Ingrediente {
        logger.debug { "Controller: Se añade un ingrediente." }
        entity.validate()
        return ingredienteRepository.add(entity) ?: throw IngredienteAlreadyExistsException("${entity.id}")
    }

    fun delete(id: Int): Ingrediente {
        logger.debug { "Controller: Se elimina un ingrediente." }
        TODO("Not yet implemented")
    }

    fun update(entity: Ingrediente): Ingrediente {
        logger.debug { "Controller: Se actualiza un ingrediente." }
        TODO("Not yet implemented")
    }
}